github/enterprise_codesecurity_configurations.go

```diff
@@ -0,0 +1,232 @@
+// Can be one of: `all`, `all_without_configurations`.
+//
+// GitHub API docs: https://docs.github.com/rest/code-security/configurations#attach-an-enterprise-configuration-to-repositories
+//
+//meta:operation POST /enterprises/{enterprise}/code-security/configurations/{configuration_id}/attach
+func (s *EnterpriseService) AttachCodeSecurityConfigurationToRepositories(ctx context.Context, enterprise string, configurationID int64, scope string) (*Response, error) {
+	u := fmt.Sprintf("enterprises/%v/code-security/configurations/%v/attach", enterprise, configurationID)
+	type scopeType struct {
+		Scope string `json:"scope"`
+	}
+
+	req, err := s.client.NewRequest("POST", u, scopeType{Scope: scope})
+	if err != nil {
+		return nil, err
+	}
+	resp, err := s.client.Do(ctx, req, nil)
+	if err != nil && resp.StatusCode != http.StatusAccepted { // StatusAccepted(202) is the expected status code as job is queued for processing
+		return resp, err
+	}
+	return resp, nil
+}
+
+// SetDefaultCodeSecurityConfiguration sets a code security configuration as a default for an enterprise.
+// `defaultForNewRepos` specifies which types of repository this security configuration should be applied to by default.
+// Can be one of: `all`, `none`, `private_and_internal`, `public`.
+//
+// GitHub API docs: https://docs.github.com/rest/code-security/configurations#set-a-code-security-configuration-as-a-default-for-an-enterprise
+//
+//meta:operation PUT /enterprises/{enterprise}/code-security/configurations/{configuration_id}/defaults
+func (s *EnterpriseService) SetDefaultCodeSecurityConfiguration(ctx context.Context, enterprise string, configurationID int64, defaultForNewRepos string) (*CodeSecurityConfigurationWithDefaultForNewRepos, *Response, error) {
+	u := fmt.Sprintf("enterprises/%v/code-security/configurations/%v/defaults", enterprise, configurationID)
+	type configParam struct {
+		DefaultForNewRepos string `json:"default_for_new_repos"`
+	}
+
+	req, err := s.client.NewRequest("PUT", u, configParam{DefaultForNewRepos: defaultForNewRepos})
+	if err != nil {
+		return nil, nil, err
+	}
+	var config *CodeSecurityConfigurationWithDefaultForNewRepos
+	resp, err := s.client.Do(ctx, req, &config)
+	if err != nil {
+		return nil, resp, err
+	}
+	return config, resp, nil
+}
+
+// ListRepositoriesForCodeSecurityConfiguration lists the repositories associated with an enterprise code security configuration.
+//
+// GitHub API docs: https://docs.github.com/rest/code-security/configurations#get-repositories-associated-with-an-enterprise-code-security-configuration
+//
+//meta:operation GET /enterprises/{enterprise}/code-security/configurations/{configuration_id}/repositories
+func (s *EnterpriseService) ListRepositoriesForCodeSecurityConfiguration(ctx context.Context, enterprise string, configurationID int64, opts *ListRepositoriesForCodeSecurityConfigurationOptions) ([]*RepositoryAttachment, *Response, error) {
```

Comment:
This endpoint returns repositories, that's why we should name it:

```suggestion
func (s *EnterpriseService) ListCodeSecurityConfigurationRepositories(ctx context.Context, enterprise string, configurationID int64, opts *ListRepositoriesForCodeSecurityConfigurationOptions) ([]*RepositoryAttachment, *Response, error) {
```